megamek/src/megamek/client/bot/princess/Princess.java

```diff
@@ -1626,10 +1626,12 @@ private void checkForBrokenEnemies() {
         }
     }
     
-    /** Update the various state trackers for a specific entity.
-     * Useful to call when receiving an entity update packet */
-    public void updateEntityState(Entity entity) {
-        if(entity.getOwner().isEnemyOf(getLocalPlayer())) {
+    /**
+     * Update the various state trackers for a specific entity.
+     * Useful to call when receiving an entity update packet
+     */
+    public void updateEntityState(final @Nullable Entity entity) {
+        if ((entity != null) && entity.getOwner().isEnemyOf(getLocalPlayer())) {
```

Comment:
My favorite implication here is that it's possible for the server to send an update for a null entity.

Reply:
I was surprised to learn this was the case, but it is according to our internals... and the log I'm working with (which also hits some of the potato fixes you made @NickAragua)